components/script_bindings/domstring.rs

```diff
@@ -26,12 +26,24 @@ use style::str::HTML_SPACE_CHARACTERS;
 use crate::script_runtime::JSContext as SafeJSContext;
 use crate::trace::RootedTraceableBox;
 
+const ASCII_END: u8 = 0x7E;
+const ASCII_CAPITAL_A: u8 = 0x41;
+const ASCII_CAPITAL_Z: u8 = 0x5A;
+const ASCII_LOWERCASE_A: u8 = 0x61;
+const ASCII_LOWERCASE_Z: u8 = 0x7A;
+const ASCII_TAB: u8 = 0x09;
+const ASCII_NEWLINE: u8 = 0x0A;
+const ASCII_FORMFEED: u8 = 0x0C;
+const ASCII_CR: u8 = 0x0D;
+const ASCII_SPACE: u8 = 0x20;
+
 #[derive(Debug, PartialEq, Eq)]
 /// A type representing the underlying encoded bytes. Either Latin1 or Utf8.
 pub enum EncodedBytes<'a> {
     /// These bytes are Latin1 encoded.
     Latin1Bytes(&'a [u8]),
-    /// This is a normal utf8 string.
+    /// This is a normal utf8 string given in bytes.
+    /// `str::from_utf8_unchecked` should be safe.
```

Comment:
Unfortunately we can't make that assumption because the type is public, so any code could create a EncodedBytes value containing invalid bytes for a UTF-8 string.